numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -685,7 +752,16 @@ PyArray_ConcatenateInto(PyObject *op, int axis, PyArrayObject *ret)
 NPY_NO_EXPORT PyObject *
 PyArray_Concatenate(PyObject *op, int axis)
 {
-    return PyArray_ConcatenateInto(op, axis, NULL);
+    /* retain legacy behaviour for casting */
```

Comment:
i was wondering when is this code path hit ? is it for 
```
np.concatenate(([0.], np.float64([1.])), out=np.empty(2, dtype=np.int64), axis=None)
```
if so maybe worth adding a test for the same ?

Reply:
This is the C-API unfortunately, which is why I just kept it as it was, since I cannot add an argument easily. But yeah, maybe I should add a test.

Reply:
Should I add a C-side test for this? And if, is that the last thing remaining here?